### DISCUSSION TITLE
Add Sprig to the golang templating

### DISCUSSION THREAD

**Discussion by spirilis** [prometheus/alertmanager#3426] [Ideas]
Can we add Sprig to the golang templating for the templates?  I'm addicted to Sprig's suite of functions from using Helm, and I'm missing it a bit right now.

Sprig is easy to use:

https://pkg.go.dev/github.com/Masterminds/sprig/v3

Example from one of my test harnesses:

Import-
```
import (
	"os"
	"testing"
	"text/template"

	sprig "github.com/Masterminds/sprig/v3"
)
```

Including it in the template chain:

```
tp := template.Must(template.New("test").Funcs(sprig.FuncMap()).Parse(testTemp))
```

Example template:

```
var testTemp = `
{
	[
		{{range .Items -}}
		{{ . | upper }}
		{{end}}
	]
}
`
```

Output:
```
go test -run BasicTemplate

{
        [
                STR1
                STR2
                STR3
                STR4

        ]
}
PASS
```

**Comment by luka5** [comment]
Even if it was already argued against introducing it in 2018 in https://github.com/prometheus/alertmanager/pull/1651, I would also like to use it in Alertmanager. It's especially useful for those who are currently migrating from [prometheus-msteams](https://github.com/prometheus-msteams/prometheus-msteams) to the new built-in Alertmanager MS Teams (thanks for that!).

**Comment by grobinson-grafana** [comment]
There is a discussion on this [here](https://github.com/prometheus/alertmanager/issues/3726).
